### PR TITLE
Feature : can switch in or out mock endpoint

### PR DESCRIPTION
**Pull request by Robert-Louis**
## PR Type

What kind of change does this PR introduce?

- Feature : Can trigger the mocked endpoint using the querystring
- Other : small cleanup
 
## What is the current behavior?

Could not switch to mocked endpoint

## What is the new behavior?

Can switch to mocked endpoint using querystring ?mock=true

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
 # The Uno Platform ToDo reference App
 
-Staging builds are available here: https://happy-moss-09a619b10.1.azurestaticapps.net
\ No newline at end of file
+Staging builds are available here: https://happy-moss-09a619b10.1.azurestaticapps.net
+
+Can trigger the mocked endpoint using the querystring ?mock=true
```
Comment (agneszitte-nventive) [inline README.md:5]:
```suggestion
- Staging builds are available here: https://happy-moss-09a619b10.1.azurestaticapps.net
- Staging builds with mocked endpoint are available here: https://happy-moss-09a619b10.1.azurestaticapps.net/?mock=true
```

File: src/ToDo.UI/Extensions/NameValueCollectionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Specialized;
+
+namespace ToDo.Extensions
+{
+    public static class NameValueCollectionExtensions
```
Comment (agneszitte-nventive) [inline src/ToDo.UI/Extensions/NameValueCollectionExtensions.cs:5]: This NameValueCollectionExtensions is used for what exactly for the app?

**Review by agneszitte-nventive — COMMENTED** [review]
@Robert-Louis you will need to commit with your GitHub user account for the CLA to work

**Comment by agneszitte-nventive** [comment]
@Robert-Louis if you can also the title of your PR to resume your changes please ;)
image

File: src/ToDo.UI/Extensions/NameValueCollectionExtensions.cs (hunk 2 above)
Comment (Robert-Louis) [reply src/ToDo.UI/Extensions/NameValueCollectionExtensions.cs:5]: in App.xaml.host.cs as NameValueCollection doesn't have a tryget

**Comment by carldebilly** [comment]
https://happy-moss-09a619b10-277.centralus.1.azurestaticapps.net/?mock=true it works!

**Comment by agneszitte-nventive** [comment]
> https://happy-moss-09a619b10-277.centralus.1.azurestaticapps.net/?mock=true it works!

yes @carldebilly but there is no taks, list

File: src/ToDo.UI/ToDo.UI.shproj [hunk 3]
```diff
@@ -10,4 +10,7 @@
   <PropertyGroup />
   <Import Project="ToDo.UI.projitems" Label="Shared" />
   <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\CodeSharing\Microsoft.CodeSharing.CSharp.targets" />
+  <ItemGroup>
+    <_Globbed_Compile Remove="Extensions\NameValueCollectionExtensions.cs" />
+  </ItemGroup>
```
Comment (carldebilly) [inline src/ToDo.UI/ToDo.UI.shproj:15]: Not useful

File: src/ToDo/Business/Models/TaskList.cs [hunk 4]
```diff
@@ -36,4 +36,6 @@ internal TaskList(TaskListData data)
 	public bool IsShared { get; init; }
 
 	public string? WellknownListName { get; init; }
+
+	public bool IsCustomListName => WellknownListName is null or "none";
```
Comment (carldebilly) [inline src/ToDo/Business/Models/TaskList.cs:40]: Is it related to this PR?

File: src/ToDo.UI/App.xaml.host.cs [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
-
 #pragma warning disable 109 // Remove warning for Window property on iOS
 
+using ToDo.Extensions;
 
 namespace ToDo;
 
 public sealed partial class App : Application
 {
+	private const string _mock = "mock";
```
Comment (carldebilly) [inline src/ToDo.UI/App.xaml.host.cs:9]: Usually constants are getting ALL_UPPERCASE.

**Review by carldebilly — APPROVED** [review]

File: src/ToDo.UI/App.xaml.host.cs [hunk 6]
```diff
@@ -20,6 +17,16 @@ private static IHost BuildAppHost()
 		var useMocks = false;
 #endif
 
+#if __WASM__
+		var stringUri = WebAssemblyRuntime.InvokeJS("window.location.href;");
+		var query = new Uri(stringUri).Query;
+		var queriesValues = System.Web.HttpUtility.ParseQueryString(query);
+
+		if (queriesValues.TryGetValue(_mock, out var value) && bool.TryParse(value, out var isMocked))
```
Comment (Xiaoy312) [inline src/ToDo.UI/App.xaml.host.cs:25]:
```suggestion
		if (queriesValues["mock"] is string mock && mock?.ToLower() == "true")
```
or better use mock=1, and just compare "1" here without tolower

File: src/ToDo.UI/Extensions/NameValueCollectionExtensions.cs (hunk 2 above)
Comment (Xiaoy312) [reply src/ToDo.UI/Extensions/NameValueCollectionExtensions.cs:5]: dont really need that

**Review by carldebilly — APPROVED** [review]

**Review by carldebilly — APPROVED** [review]

**Comment by nickrandolph** [comment]
I know I'm late to the party but this PR breaks the default behaviour of running with mocks in debug - I'll add the #if DEBUG back to the App.host.cs for convenience in a separate PR

**Pull request merged**